server/blob/blob-api/src/test/java/org/apache/james/blob/api/BlobStoreContract.java

```diff
@@ -175,4 +179,56 @@ default void readShouldReturnBigSavedData() {
 
         assertThat(read).hasSameContentAs(new ByteArrayInputStream(TWELVE_MEGABYTES));
     }
+
+    @Test
+    @Disabled("JAMES-2829 Not supported yet")
+    default void deleteShouldNotThrowWhenBlobDoesNotExist() {
+        assertThatCode(() -> testee().delete(testee().getDefaultBucketName(), blobIdFactory().randomId()).block())
+            .doesNotThrowAnyException();
+    }
+
+    @Test
+    @Disabled("JAMES-2829 Not supported yet")
+    default void deleteShouldDeleteExistingBlobData() {
+        BlobId blobId = testee().save(testee().getDefaultBucketName(), SHORT_BYTEARRAY).block();
+        testee().delete(testee().getDefaultBucketName(), blobId).block();
+
+        assertThatThrownBy(() -> testee().read(testee().getDefaultBucketName(), blobId))
+            .isInstanceOf(ObjectStoreException.class);
+    }
+
+    @Test
+    @Disabled("JAMES-2829 Not supported yet")
+    default void deleteShouldBeIdempotent() {
+        BlobId blobId = testee().save(testee().getDefaultBucketName(), SHORT_BYTEARRAY).block();
+        testee().delete(testee().getDefaultBucketName(), blobId).block();
+
+        assertThatCode(() -> testee().delete(testee().getDefaultBucketName(), blobId).block())
+            .doesNotThrowAnyException();
+    }
+
+    @Test
+    @Disabled("JAMES-2829 Not supported yet")
+    default void deleteShouldNotDeleteOtherBlobs() {
+        BlobId blobIdToDelete = testee().save(testee().getDefaultBucketName(), SHORT_BYTEARRAY).block();
+        BlobId otherBlobId = testee().save(testee().getDefaultBucketName(), ELEVEN_KILOBYTES).block();
+
+        testee().delete(testee().getDefaultBucketName(), blobIdToDelete).block();
+
+        InputStream read = testee().read(testee().getDefaultBucketName(), otherBlobId);
+
+        assertThat(read).hasSameContentAs(new ByteArrayInputStream(ELEVEN_KILOBYTES));
+    }
+
+    @Test
+    @Disabled("JAMES-2829 Not supported yet")
+    default void deleteConcurrentlyShouldNotFail() throws Exception {
```

Comment:
while you are at it, could you check that you read either the entire blob or nothing but you don't have partial reads?